_posts/2024-01-04-glibc-xcrypt.md

```diff
@@ -0,0 +1,42 @@
+---
+title: glibc 2.38 Update Issues and Solutions
+layout: post
+---
+
+With the update to glibc 2.38, `libcrypt.so.1` is no longer part of glibc. This
+is an important library providing cryptographic functions for some core system
+packages like `pam`. The library is now `libcrypt.so.2` and is provided by
+`libxcrypt`. If the library `libcrypt.so.1` is still needed by an application
+(e.g. a precompiled or proprietary one), `libxcrypt-compat` provides it.
+
+With this change, some kinds of partial upgrades can leave PAM unable to
+function. This breaks tools like `sudo`, `doas`, and `su`, as well as breaking
+authentication to your system. If this has happened to you, you can either:
+
+- add `init=/bin/sh` to your kernel command-line in the bootloader and
+   [downgrade](https://docs.voidlinux.org/xbps/advanced-usage.html#downgrading)
+   glibc,
+- or mount the system's root partition in a live environment,
+   [chroot](https://docs.voidlinux.org/config/containers-and-vms/chroot.html#chroot-usage)
+   into it, and install `libxcrypt-compat`
+
+This should allow you to access your system normally and run a full update.
```

Comment:
Either of these steps should allow you to access your system as normal and run a full update.